src/vi/tranh18/src/eu/kanade/tachiyomi/extension/vi/tranh18/Tranh18.kt

```diff
@@ -0,0 +1,213 @@
+package eu.kanade.tachiyomi.extension.vi.tranh18
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import kotlin.collections.sortedByDescending
+
+class Tranh18 : ParsedHttpSource() {
+    override val lang: String = "vi"
+
+    override val name: String = "Tranh18"
+
+    override val baseUrl: String = "https://tranh18z.com"
+
+    override val supportsLatest: Boolean = true
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .rateLimit(3)
+        .build()
+
+    override fun headersBuilder(): Headers.Builder = super.headersBuilder()
+        .add("Referer", "$baseUrl/")
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/update" + if (page > 1) "?page=$page" else "", headers)
+    }
+
+    override fun latestUpdatesSelector(): String = ".box-body ul li, .manga-list ul li"
+
+    override fun latestUpdatesFromElement(element: Element): SManga = SManga.create().apply {
+        val sel = element.select(".mh-item, .manga-list-2-cover")
+        setUrlWithoutDomain(sel.select("a").attr("href"))
+        title = sel.select("a").attr("title")
+        thumbnail_url = baseUrl + sel.select("p.mh-cover").attr("style")
+            .substringAfter("url(")
+            .substringBefore(")")
+            .ifEmpty { baseUrl + sel.select("img").attr("data-original") }
+    }
+
+    override fun latestUpdatesNextPageSelector(): String = ".mt20"
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET(baseUrl, headers)
+    }
+
+    override fun popularMangaFromElement(element: Element): SManga = latestUpdatesFromElement(element)
+
+    override fun popularMangaSelector(): String = latestUpdatesSelector()
+
+    override fun popularMangaNextPageSelector(): String = latestUpdatesNextPageSelector()
+
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        val infoMobile = document.select(".detail-main")
+        title = document.select(".info h1").takeIf { it.isNotEmpty() }
+            ?.text()
+            ?: infoMobile.select(".detail-main-info-title").text()
```

Comment:
These selectors can also be combined
```css
.info h1, .detail-main-info-title
```

probably other fields as well. I see this pattern is being repeated.